src/main/kotlin/org/jenkinsci/gradle/plugins/accmod/CheckAccessModifierTask.kt

```diff
@@ -0,0 +1,65 @@
+package org.jenkinsci.gradle.plugins.accmod
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.file.FileCollection
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.*
+import org.gradle.kotlin.dsl.listProperty
+import org.gradle.kotlin.dsl.mapProperty
+import org.gradle.kotlin.dsl.property
+import org.gradle.kotlin.dsl.submit
+import org.gradle.workers.WorkerExecutor
+import java.io.File
+import javax.inject.Inject
+
+/**
+ * This task is modeled on org.kohsuke:access-modifier-checker
+ *
+ * @see org.kohsuke.accmod.impl.EnforcerMojo
+ */
+open class CheckAccessModifierTask @Inject constructor(private val workerExecutor: WorkerExecutor) : DefaultTask() {
+    companion object {
+        const val NAME = "checkAccessModifier"
+        const val PREFIX = "$NAME."
+    }
+
+    @Classpath
+    val accessModifierClasspath: Property<Configuration> = project.objects.property()
+
+    @Input
+    val accessModifierProperties: MapProperty<String, Any> = project.objects.mapProperty()
+
+    @CompileClasspath
+    val compileClasspath: Property<Configuration> = project.objects.property()
+
+    @InputFiles
+    val compilationDirs: ListProperty<File> = project.objects.listProperty()
+
+    @Internal
+    val presentCompilationDirs: Provider<List<File>> = compilationDirs.map { it.filter { f -> f.exists() } }
+
+    @Internal
+    val classpathToScan: Provider<FileCollection> = compileClasspath.map {
+        val dependencies = it.resolvedConfiguration.resolvedArtifacts.map { a -> a.file }
+        val compiled = presentCompilationDirs.get()
+        project.layout.files(compiled, dependencies)
+    }
+
+    @TaskAction
+    fun check() {
+        val q = workerExecutor.classLoaderIsolation {
+            classpath.from(accessModifierClasspath)
+        }
+        for (compilationDir in presentCompilationDirs.get()) {
+            q.submit(CheckAccess::class) {
```

Comment:
An alternative to using a worker here (a that's what you were asking: https://gradle-community.slack.com/archives/CA745PZHN/p1608574892315400) would be to use an `URLClassLoader`, but because you're using a for loop here, the worker will actually parallelize work so I think it's worth it.

Reply:
I think I'll stick with the worker api, but just to make sure I'm understanding the alternative is:

1. Store the current classloader in a variable
2. Update the current classloader to a new `URLClassLoader`
3. Run the task
4. Swap back to the original classloader

Reply:
I think it'd be more complicated: you'd have to have the task code (that calls Kohsuke's lib) in the same class loader as the lib (not even a parent classloader, possibly a child) which generally involves writing your own classloader (I've done that a couple time, it's always a PITA; Maven's ClassWorlds make it easier; Gradle has something equivalent but it's internal).
Either that or exclusively using reflection (or if you switch the current classloader, then afaict only use classes from the lib or the jdk until you switch back).

In other words: stick to the worker API 😉

Reply:
Thanks for the explanation